src/connection.c

```diff
@@ -83,333 +82,460 @@ void set_global_reconnect_status(bool status)
+    jwt_server->allow_insecure = allow_insecure_conn(
+             jwt_server->server_addr, SERVER_ADDR_LEN,
+             jwt_server->port, PORT_LEN);
+    if (jwt_server->allow_insecure < 0) {
+      set_server_null (jwt_server);
+      return FIND_JWT_FAIL;
+    }
+    ParodusInfo("JWT ON: jwt_server_url stored as %s\n", jwt_server->server_addr);
+  }
+#endif
+  return 0;
+}
+
+
+//--------------------------------------------------------------------
+// connect to current server
+static int nopoll_connect (create_connection_ctx_t *ctx, int is_ipv6)
+{
+   noPollCtx *nopoll_ctx = ctx->nopoll_ctx;
+   server_t *server = ctx->current_server;
+   noPollConn *connection;
+   noPollConnOpts * opts;
+   char *default_url = get_parodus_cfg()->webpa_path_url; 
+
+   if (server->allow_insecure > 0) {
+      ParodusPrint("secure false\n");
+      opts = createConnOpts(ctx->extra_headers, false); 
+      connection = nopoll_conn_new_opts (nopoll_ctx, opts, 
+        server->server_addr,server->port,
+        NULL, default_url,NULL,NULL);// WEBPA-787
+   } else {
+      ParodusPrint("secure true\n");
+      opts = createConnOpts(ctx->extra_headers, true);
+      if (is_ipv6) {
+         ParodusInfo("Connecting in Ipv6 mode\n");
+         connection = nopoll_conn_tls_new6 (nopoll_ctx, opts, 
+           server->server_addr, server->port,
+           NULL, default_url,NULL,NULL);
+      } else {      
+         ParodusInfo("Connecting in Ipv4 mode\n");
+         connection = nopoll_conn_tls_new (nopoll_ctx, opts, 
+           server->server_addr, server->port,
+           NULL, default_url,NULL,NULL);
+      }      
+   }
+   if (NULL == connection) {
+     if((checkHostIp(server->server_addr) == -2)) {
+       if (check_timer_expired (&ctx->connect_timer, 15*60*1000)) {
+  	 ParodusError("WebPA unable to connect due to DNS resolving to 10.0.0.1 for over 15 minutes; crashing service.\n");
+	 set_global_reconnect_reason("Dns_Res_webpa_reconnect");
+	 set_global_reconnect_status(true);
+						
+	 kill(getpid(),SIGTERM);						
+       }
+     }
+   }
+           
+   set_global_conn(connection);
+   return (NULL != connection);
+}
+
+//--------------------------------------------------------------------
+// Return codes for wait_connection_ready
+#define WAIT_SUCCESS	0
+#define WAIT_ACTION_RETRY	1	// if wait_status is 307, 302, 303 or 403
+#define WAIT_FAIL 	2
+
+static int wait_connection_ready (create_connection_ctx_t *ctx)
+{
+  int wait_status;
+  char redirectURL[128]={'\0'};
+
+  if(nopoll_conn_wait_until_connection_ready(get_global_conn(), 10, 
+	&wait_status, redirectURL)) 
+     return WAIT_SUCCESS;
+  if(wait_status == 307 || wait_status == 302 || wait_status == 303)    // only when there is a http redirect
+  {
+	char *redirect_ptr = redirectURL;
+	ParodusError("Received temporary redirection response message %s\n", redirectURL);
+	// Extract server Address and port from the redirectURL
+	if (strncmp (redirect_ptr, "Redirect:", 9) == 0)
```

Comment:
No need for `strncmp()` here since you are passing in a fixed/static string and comparing all of it.

Reply:
ONly comparing the front, the first 9 characters.

Reply:
ok